app/com/arpnetworking/kairos/client/models/MetricsQuery.java

```diff
@@ -0,0 +1,547 @@
+    }
+
+    private final DateTime _startTime;
+    private final DateTime _endTime;
+    private final List<Metric> _metrics;
+
+    /**
+     * Implementation of the builder pattern for MetricsQuery.
+     */
+    public static final class Builder extends OvalBuilder<MetricsQuery> {
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            super(MetricsQuery::new);
+        }
+
+        /**
+         * Sets the start time of the query. Required. Cannot be null.
+         *
+         * @param value the start time
+         * @return this {@link Builder}
+         */
+        public Builder setStartTime(final DateTime value) {
+            _startTime = value;
+            return this;
+        }
+
+        /**
+         * Sets the end time of the query. Optional. Cannot be null.  Default is now.
+         *
+         * @param value the end time
+         * @return this {@link Builder}
+         */
+        public Builder setEndTime(final DateTime value) {
+            _endTime = value;
+            return this;
+        }
+
+        /**
+         * Sets list of metrics. Required. Cannot be null or empty.
+         *
+         * @param value the metrics
+         * @return this {@link Builder}
+         */
+        public Builder setMetrics(final List<Metric> value) {
+            _metrics.clear();
+            _metrics.addAll(value);
+            return this;
+        }
+
+        /**
+         * Adds a metric to the list of metrics. Cannot be null.
+         *
+         * @param value a metric
+         * @return this {@link Builder}
+         */
+        public Builder addMetric(final Metric value) {
+            _metrics.add(value);
+            return this;
+        }
+
+        /**
+         * Creates a new builder based on this one.
+         *
+         * @return a new builder with the same
+         */
+        public Builder copy() {
+            return new Builder().setStartTime(_startTime)
+                    .setEndTime(_endTime)
+                    .setMetrics(_metrics.stream()
+                            .map(Metric.Builder::from)
+                            .map(Metric.Builder::build)
+                            .collect(Collectors.toList()));
+        }
+
+        public DateTime getStartTime() {
+            return _startTime;
+        }
+
+        public DateTime getEndTime() {
+            return _endTime;
+        }
+
+        public List<Metric> getMetrics() {
+            return _metrics;
+        }
+
+        @NotNull
+        private DateTime _startTime;
+        @NotNull
+        private DateTime _endTime = DateTime.now();
+        @NotNull
+        @NotEmpty
+        private List<Metric> _metrics = Lists.newArrayList();
+    }
+
+    /**
+     * Holds the data for a Metric element of the query.
+     *
+     * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+     */
+    public static final class Metric {
```

Comment:
Should we break these out into top level classes? It seems like they have an appropriate package.

Reply:
I don't think so.  These are all types that are embedded in the metrics data query; inner classes for the inner json objects.